ckanext/unaids/tests/test_dataset_transfer.py

```diff
@@ -0,0 +1,66 @@
+'Tests for plugin.py.'
+# encoding: utf-8
+
+from ckan.tests import helpers
+import ckan.model as model
+from ckan.lib.helpers import url_for
+from ckan.tests import factories
+from ckan.plugins import toolkit
+import pytest
+import logging
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.ckan_config('ckan.plugins', 'unaids scheming_datasets')
+@pytest.mark.usefixtures('with_plugins')
+@pytest.mark.usefixtures('clean_db')
+class TestDatasetTransfer(object):
+
+    def test_dataset_transfer_request(self, app):
+
+        # create 3 users and orgs
+        user_1, user_2, user_3 = [
+            factories.User()
+            for x in range(3)
+        ]
+        org_1, org_2, org_3 = [
+            factories.Organization(user=user)
+            for user in [user_1, user_2, user_3]
+        ]
+
+        # create an org_1 dataset pending transfer to org_2
+        dataset = factories.Dataset(
+            owner_org=org_1['id'],
+            type='test-schema',
+            org_to_allow_transfer_to=org_2['id']
+        )
+        transfer_dataset_url = url_for(
+            'unaids_dataset_transfer.process_dataset_transfer',
+            dataset_id=dataset['id']
+        )
+
+        # user_1 and user_3 should not be able to accept
+        # the transfer as they are not members of org_2
+        for user in [user_1, user_3]:
+            response = app.get(
+                url=transfer_dataset_url,
+                extra_environ={'REMOTE_USER': user['name']},
+                status=403 # expect forbidden error
+            )
```

Comment:
So are actually asserting here that a 403 is raised?  Could be good to do this using more standardised pytest syntax: 
```suggestion
        from werkzeug.exceptions import Forbidden
        with pytest.raises(Forbidden):
            response = app.get( 
                url=transfer_dataset_url,
￼                extra_environ={'REMOTE_USER': user['name']}
            )
```

Reply:
Import the exception at the top of the file obviously.

Reply:
I copied this style of writing the test off another test elsewhere.
When expecting a 200 status I get an `AssertionError` rather than `Forbidden` which is equally as informative

```
for user in [user_1, user_3]:
    with pytest.raises(AssertionError):
        app.get(
            url=transfer_dataset_url,
            extra_environ={"REMOTE_USER": user["name"]},
            status=200,  # should return a 403
        )
```

It feels like sticking with my original `status=403 # expect forbidden error` code is more descriptive.
What do you think?

Reply:
I'd probably do...